src/NServiceBus.Core/Recoverability/FirstLevelRetries/FirstLevelRetries.cs

```diff
@@ -44,5 +51,35 @@ int GetMaxRetries(ReadOnlySettings settings)
 
         }
 
+        class FlrStatusStorageCleaner : FeatureStartupTask
+        {
+            public FlrStatusStorageCleaner(FlrStatusStorage statusStorage)
+            {
+                this.statusStorage = statusStorage;
+            }
+
+            protected override void OnStart()
+            {
+                timer = new Timer(ClearFlrStatusStorage, null, ClearingInterval, ClearingInterval);
+            }
+
+            protected override void OnStop()
+            {
+                if (timer != null)
+                {
+                    timer.Dispose();
```

Comment:
I don't think that's necessary. There's a chance that the timer callback gets invoked after the dispose anyway and I don't see an issue there since we don't remove the storage.